Content.Server/Sticky/Systems/StickySystem.cs

```diff
@@ -20,15 +21,40 @@ public sealed class StickySystem : EntitySystem
     [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
     [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
     [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
 
     private const string StickerSlotId = "stickers_container";
 
     public override void Initialize()
     {
         base.Initialize();
+
+        SubscribeLocalEvent<StickyComponent, MapInitEvent>(OnMapInit);
         SubscribeLocalEvent<StickyComponent, StickyDoAfterEvent>(OnStickFinished);
         SubscribeLocalEvent<StickyComponent, AfterInteractEvent>(OnAfterInteract);
         SubscribeLocalEvent<StickyComponent, GetVerbsEvent<Verb>>(AddUnstickVerb);
+
+        SubscribeLocalEvent<TransferStickySurfaceOnSpawnBehaviorComponent, DestructionSpawnBehavior>(OnDestructionSpawnBehavior);
+    }
+
+    private void OnMapInit(EntityUid uid, StickyComponent component, MapInitEvent args)
+    {
+        if (!component.StickOnStart || component.StuckTo != null)
+            return;
+
+        var stuck = false;
+        foreach (var entity in _lookup.GetEntitiesIntersecting(Transform(uid).MapPosition))
```

Comment:
You shouldn't be getting transform up to map and then back down to local as it's far more expensive and unreliable than it needs to be, use a different method.